File: Leetcode34.py
class Solution(object):
    def searchRange(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        
        
        left = self.binarysearch(nums, target, True)
        right = self.binarysearch(nums, target, False)
        return (left,right)
        
        
        
    def binarysearch(self, nums, target, IsLeft):
            
        l = 0
        r = len(nums) - 1
        i = -1
        while l <= r:
            mid = (l + r) // 2
            if nums[mid] > target:
                r = mid - 1
            elif nums[mid] < target:
                l = mid + 1
            else:
                i = mid
                if IsLeft:
                    r = mid - 1
                else:
                    l = mid + 1
        return i
        

File: Leetcode34.js
Time complexity - O(logn)
Space complexity - logn.

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var searchRange = function(nums, target) {
    if(!nums.includes(target)){
        return [-1,-1];
    }
    var left = leftside(nums,target);
    var rgt  = rightside(nums,target);
    return [rgt,left];
 }
var leftside = function(nums,target){
    var left = -1,right=nums.length;
    while(right-left>1){
        var mid = Math.floor((right+left)/2);
        if(nums[mid]<=target){
            left = mid;
        }else{
            right = mid;
        }
    }
    return right-1;
}
var rightside = function(nums,target){
    var left=-1,right=nums.length;
    while(right-left>1){
        var mid = Math.floor((left+right)/2);
        if(nums[mid]<target){
         left = mid;
        }
        else{
         right = mid;
        }
    }
    return left+1;
}
